doc/man7/provider-kdf.pod

```diff
@@ -0,0 +1,325 @@
+
+Sets the number of iterations in the associated KDF ctx.
+
+=item "mode" (B<OSSL_KDF_PARAM_MODE>) <UTF8 string>
+
+Sets the mode in the associated KDF ctx.
+
+=item "pkcs5" (B<OSSL_KDF_PARAM_PKCS5>) <integer>
+
+Enables or diables the SP800-132 compliance checks.
+A mode of 0 enables the compliance checks.
+
+The checks performed are:
+
+=over 4
+
+=item - the iteration count is at least 1000.
+
+=item - the salt length is at least 128 bits.
+
+=item - the derived key length is at least 112 bits.
+
+=back
+
+=item "ukm" (B<OSSL_KDF_PARAM_UKM>) <octet string>
+
+Sets an optional random string that is provided by the sender called
+"partyAInfo".  In CMS this is the user keying material.
+
+
+=item "cekalg" (B<OSSL_KDF_PARAM_CEK_ALG>) <UTF8 string>
+
+Sets the CEK wrapping algorithm name in the associated KDF ctx.
+
+=item "n" (B<OSSL_KDF_PARAM_SCRYPT_N>) <unsigned integer>
+
+Sets the scrypt work factor parameter N in the associated KDF ctx.
+
+=item "r" (B<OSSL_KDF_PARAM_SCRYPT_R>) <unsigned integer>
+
+Sets the scrypt work factor parameter r in the associated KDF ctx.
+
+=item "p" (B<OSSL_KDF_PARAM_SCRYPT_P>) <unsigned integer>
+
+Sets the scrypt work factor parameter p in the associated KDF ctx.
+
+=item "maxmem_bytes" (B<OSSL_KDF_PARAM_SCRYPT_MAXMEM>) <unsigned integer>
+
+Sets the scrypt work factor parameter maxmem in the associated KDF ctx.
+
+=item "info" (B<OSSL_KDF_PARAM_INFO>) <octet string>
+
+Sets the optional shared info in the associated KDF ctx.
+
+=item "seed" (B<OSSL_KDF_PARAM_SEED>) <octet string>
+
+Sets the IV in the associated KDF ctx.
+
+=item "xcghash" (B<OSSL_KDF_PARAM_SSHKDF_XCGHASH>) <octet string>
+
+Sets the xcghash in the associated KDF ctx.
+
+=item "session_id" (B<OSSL_KDF_PARAM_SSHKDF_SESSION_ID>) <octet string>
+
+Sets the session ID in the associated KDF ctx.
+
+=item "type" (B<OSSL_KDF_PARAM_SSHKDF_TYPE>) <integer>
+
+Sets the SSH KDF type parameter in the associated KDF ctx.
+There are six supported types:
+
+=over 4
+
+=item EVP_KDF_SSHKDF_TYPE_INITIAL_IV_CLI_TO_SRV
+
+The Initial IV from client to server.
+A single char of value 65 (ASCII char 'A').
+
+=item EVP_KDF_SSHKDF_TYPE_INITIAL_IV_SRV_TO_CLI
+
+The Initial IV from server to client
+A single char of value 66 (ASCII char 'B').
+
+=item EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_CLI_TO_SRV
+
+The Encryption Key from client to server
+A single char of value 67 (ASCII char 'C').
+
+=item EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_SRV_TO_CLI
+
+The Encryption Key from server to client
+A single char of value 68 (ASCII char 'D').
+
+=item EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_CLI_TO_SRV
+
+The Integrity Key from client to server
+A single char of value 69 (ASCII char 'E').
+
+=item EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_SRV_TO_CLI
+
+The Integrity Key from client to server
+A single char of value 70 (ASCII char 'F').
+
+=back
+
+=item "size" (B<OSSL_KDF_PARAM_SIZE>) <unsigned integer>
+
+Gets the output size from the associated KDF ctx.
```

Comment:
Maybe add some comments on using SIZE_MAX if the algorithm supports variable output size. Citing the EVP_KDF_CTX_get_kdf_size manpage:
`EVP_KDF_CTX_get_kdf_size() returns the output size if the algorithm produces a fixed amount of output and SIZE_MAX otherwise. If an error occurs then 0 is returned. For some algorithms an error may result if input parameters necessary to calculate a fixed output size have not yet been supplied.`

Also perhaps put it as last or to a separate list as this is the only gettable parameter.

Reply:
I've put it first, so it isn't missed so easily.  I've changed the text to:
> Gets the output size from the associated KDF ctx.  If the algorithm produces a variable amount of output, SIZE_MAX should be returned.  If the input parameters required to calculate the fixed output size have not yet been supplied, 0 should be returned indicating an error.